### PR TITLE
Android responsive overflow issues

### LINKED ISSUE #8380: Android App - Responsive design issues
## Observed behavior
There are many screens with text or button layout issues when the app is viewed in a mobile phone.

## Expected behavior
All pages should be optimized for mobile devices.

## Steps to reproduce the issue
1. Install the Android [build](https://buildkite.com/learningequality/kolibri-android-installer/builds/2548#7d02075f-02ab-40ea-8f21-aa8bd75782f2) on a mobile device.
2. Interact with the app.

## Screenshots

1 | 2 | 3
--- | --- | ---
Screenshot_20210901-103337_Kolibri | Screenshot_20210901-103350_Kolibri | Screenshot_20210901-162205_Kolibri

4 | 5 | 6
--- | --- | ---
Screenshot_20210901-162225_Kolibri | Screenshot_20210901-162727_Kolibri | Screenshot_20210901-162917_Kolibri

## Usage Details
 - OS: Android 11
 - Device: Samsung Galaxy Note 10 Lite

### PR DESCRIPTION
**Pull request by marcellamaki**
## Summary

Small improvements to layout and spacing, mainly in device set up and coach, on mobile views. 

## References
Fixes #8380 with in-kolibri fixes. If some will be made in KDS, these can be reverted.

|  1 | 2  | 
|---|---|
| Screen Shot 2021-12-17 at 12 22 29 PM| Screen Shot 2021-12-17 at 12 22 53 PM |
|  3 & 4  | 5  |
|  new-lesson-cards | Screen Shot 2021-12-17 at 12 44 30 PM |

6 has different buttons than when the issue was first created

## Reviewer guidance

- Are these acceptable for now?
- Are there any parts that are too risky as potential causes of regressions, that should just be removed?

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/assets/src/views/CoreBase/index.vue [hunk 1]
```diff
@@ -607,4 +607,9 @@
     line-height: 2em;
   }
 
+  .small {
+    max-width: 100%;
+    overflow-x: hidden !important;
+  }
```
Comment (indirectlylit) [inline kolibri/core/assets/src/views/CoreBase/index.vue]: This change strikes me as a bit risky. `!important` has a tendency to cause unexpected breakage and going on `CoreBase` the effects could be wide-ranging.

File: kolibri/plugins/coach/assets/src/views/plan/LessonResourceSelectionPage/index.vue [hunk 2]
```diff
@@ -570,6 +574,26 @@
...
 
+  .mobile-bookmark-container {
+    display: flex;
+    max-width: 100%;
+    min-height: 141px;
+    margin: auto;
+    margin-bottom: 24px;
+    border-radius: 2px;
+    box-shadow: 0 1px 5px 0 #a1a1a1, 0 2px 2px 0 #e6e6e6, 0 3px 1px -2px #ffffff;
+    transition: box-shadow 0.25s ease;
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/plan/LessonResourceSelectionPage/index.vue]: built-in easing curves!

```scss
      .ease:hover {
        @extend %dropshadow-8dp;
        @extend %md-decelerate-func;

        transition: all $core-time;
      }
```

newly documented!


https://deploy-preview-287--kolibri-design-system.netlify.app/styling/#easing

File: kolibri/plugins/coach/assets/src/views/plan/LessonResourceSelectionPage/index.vue [hunk 3]
```diff
@@ -570,6 +574,26 @@
...
   }
 
+  .mobile-bookmark-container {
+    display: flex;
+    max-width: 100%;
+    min-height: 141px;
+    margin: auto;
+    margin-bottom: 24px;
+    border-radius: 2px;
+    box-shadow: 0 1px 5px 0 #a1a1a1, 0 2px 2px 0 #e6e6e6, 0 3px 1px -2px #ffffff;
```
Comment (indirectlylit) [inline kolibri/plugins/coach/assets/src/views/plan/LessonResourceSelectionPage/index.vue]: built-in drop shadows!
```scss
        @extend %dropshadow-2dp;
```

newly documented!

https://deploy-preview-287--kolibri-design-system.netlify.app/styling/#elevation

File: kolibri/core/assets/src/views/CoreBase/index.vue (hunk 1 above)
Comment (marcellamaki) [reply kolibri/core/assets/src/views/CoreBase/index.vue]: I actually am pretty sure this is not needed and was something I was using for troubleshooting. Good catch! I'll definitely remove it, and will update the PR to indicate whether it changes any of the items described above, but I'm hoping it won't.

Comment (marcellamaki) [reply kolibri/core/assets/src/views/CoreBase/index.vue]: Manual QA seems to still check out with this removed

**Comment by radinamatic** [comment]
Overflowing outside the container is fixed, but now it has the same issue I reported sometime ago (#8808)

1 ❗  | 2 ❗ | 3 ✔️ 
--- | --- | ---
IMG_20211217_225734862 | IMG_20211217_230009815_HDR | IMG_20211217_225932109_HDR

Labels for radio buttons should also wrap to the second line, and not be truncated. Actually the example 3 is not representative, as what wraps correctly is the paragraph **below** the label, but I hope the issue is clear.

**Comment by nucleogenesis** [comment]
> Labels for radio buttons should also wrap to the second line, and not be truncated 
 
I ran into that in #8946  - I think we'll want to discuss this in KDS about the Radio Buttons. 

It is worth noting that the second one that Radina points out is the expected fix of the issue that brought this all about.

First - it was because network names could be long and wouldn't wrap #6754 so the solution was then to add a text truncation class to the KRadioButton which wouldn't be noticed anywhere in Kolibri except in the settings now. 

In any case, some deep styles removing that seems like the best step at this point

**Comment by marcellamaki** [comment]
Thanks both -- I can try to do some work-around /deep/ changes for now in these specific areas, but yes let's revisit some of this in KDS in the new year?

**Comment by radinamatic** [comment]
I'd say network names should be the edge case as the majority of labels for radio buttons are _not_ network names, but real options that user does not have any other means to infer the content from (there's no _info_ tooltip or such).

**Review by nucleogenesis — APPROVED** [review]
Code LGTM

**Comment by marcellamaki** [comment]
I rebased and this now includes the changes from @nucleogenesis's previous PR, which were incidentally the same changes I needed to make to adddress the radio button wrapping. Should be good to go!

**Pull request merged**